src/zarr/storage/_common.py

```diff
@@ -314,9 +319,16 @@ async def make_store_path(
             # We deliberate only consider dict[str, Buffer] here, and not arbitrary mutable mappings.
             # By only allowing dictionaries, which are in-memory, we know that MemoryStore appropriate.
             store = await MemoryStore.open(store_dict=store_like, read_only=_read_only)
+        elif _has_fsspec and isinstance(store_like, FSMap):
+            if path:
+                raise ValueError("'path' was provided but is not used for FSMap store_like objects")
```

Comment:
Can we add a constructive instruction to this error? Something like "...set path to `None`, or leave it unset, when specifying a store with an FSMap"